### PR TITLE
Add API server and make all the XHR requests pass through the API layer

### PR DESCRIPTION
**Pull request by nickhsine**
runtime transpilation for es6/7

add eslint ignore file

move browsersync-server to bin folder and add global variable

add webpack global variable

Update package.json
1. add dependencies
2. add custom script

Update server/server.js
1. add error handling for http proxy
2. add eslint condition

Add api layer on FE server
1. change redux action
2. add corresponding actions of api layer

Add bin/api.js and api/api.js
delete unused files

move constants variable into constants.js

@hcchien @garfieldduck

### CODE REVIEW

File: src/middleware/api.js [hunk 1]
```diff
@@ -7,6 +9,15 @@ import _ from 'lodash'
 
 export const CALL_API = Symbol('CALL_API')
 
+function formatUrl(path) {
+  if (__SERVER__) {
+    // Prepend host and port of the API server to the path.
+    return 'http://' + config.apiHost + ':' + config.apiPort + path
```
Comment (hcchien) [inline src/middleware/api.js]: would you include the protocol to the variable?

File: api/constants.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+let config = {
+  API: 'https://www.twreporter.org/api'
+}
+
+export default config
```
Comment (hcchien) [inline api/constants.js]: why don't you put them into the config.js?

Comment (nickhsine) [inline api/constants.js]: ok

File: src/middleware/api.js (hunk 1 above)
Comment (nickhsine) [inline src/middleware/api.js]: on server side, we only use ```http``` protocol.

**Pull request merged**